src/engine/v3/V3Interpreter.v3

```diff
@@ -1108,6 +1109,91 @@ class V3Interpreter extends WasmStack {
 			I32X4_TRUNC_SAT_F64X2_U_ZERO => do_v_v(V3Eval.I32X4_TRUNC_SAT_F64X2_U_ZERO);
 			I8X16_SHUFFLE => doShuffle();
 			I8X16_SWIZZLE => doSwizzle();
+			// ext: typed-continuation
+			CONT_NEW => {
+				var cont_index = codeptr.read_uleb32();
+				var cont_decl = ContDecl.!(frame.func.instance.heaptypes[cont_index]);
+				var func = Function.!(Value.Ref.!(pop()).val);
+				if (func == null) trap(TrapReason.NULL_DEREF);
+
+				var cont = Continuation.new(cont_decl, V3Interpreter.new());
+				cont.stack.reset(func);
+				push(Value.Ref(cont));
+			}
+			CONT_BIND => {
+				var in_cont = ContDecl.!(frame.func.instance.heaptypes[codeptr.read_uleb32()]);
+				var out_cont = ContDecl.!(frame.func.instance.heaptypes[codeptr.read_uleb32()]);
+				var n_binds = in_cont.sig.params.length - out_cont.sig.params.length;
+
+				var cont = Continuation.!(Value.Ref.!(pop()).val);
+				if (cont == null) trap(TrapReason.NULL_DEREF);
+				// continuation can only be resumed/bound once
+				if (cont.used) trap(TrapReason.USED_CONTINUATION);
+
+				var new_cont = Continuation.new(cont.cont, cont.stack);
+				for (i < cont.bounded.length) new_cont.bounded.put(cont.bounded[i]);
+
+				var range = values.peekn(n_binds);
+				for (i < range.length) new_cont.bounded.put(range[i]);
+				push(Value.Ref(cont));
+
+				values.top -= n_binds;
+			}
+			RESUME => {
+				var cont_idx = codeptr.read_uleb32();
+				var cont_type = ContDecl.!(frame.func.instance.heaptypes[cont_idx]);
+				var handlers = codeptr.read_handlers();
```

Comment:
This is fine for now, but I think we'll want to use the sidetable entry to directly get to the handler table. For example, the fast interpreter should avoid decoding the handlers and instead have a direct pointer to a handler table in the sidetable, as well as the length of the `resume` opcode because it will fall through to the next opcode upon return.

Reply:
Sorry I don't think I quite get the part of "having a direct pointer to a handler table". By this do you mean to cache/canonicalize the handler information somewhere else in the module during initialization, and iterate through the cached version to find the appropriate tag during `resume` (like the way `try_table` are implemented)?

Reply:
Right, similar to exception handling, there should be a handler table hanging off of the function that is searched when suspending. IIUC then the handler table for the top frame of a stack only needs to be searched upon suspend (no iteration of frames, just stacks via parent pointers). It might even be possible to use the exact same format of handler tables and share the search logic.